### PR TITLE
[HETERO] Fix segfault in supported/unsuppoterd layers check

### PR DESCRIPTION
**Pull request by olpipi**
### Details:
 - *item1*
 - *...*

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: src/plugins/intel_gpu/src/plugin/plugin.cpp [hunk 1]
```diff
@@ -454,8 +454,8 @@ QueryNetworkResult Plugin::QueryNetwork(const CNNNetwork& network,
         }
     }
 
-    for (auto&& layerName : supported) {
-        if (InferenceEngine::details::contains(unsupported, layerName)) {
+    for (auto&& layerName : unsupported) {
+        if (InferenceEngine::details::contains(supported, layerName)) {
```
Comment (ilya-lavrenov) [inline src/plugins/intel_gpu/src/plugin/plugin.cpp:458]: am I right that logic is the same, but we fix SEGFAULT which was caused that we iterate over container which is changed inside the loop?

Comment (olpipi) [reply src/plugins/intel_gpu/src/plugin/plugin.cpp:458]: yeap, you're right.
next layerName after erase call was invalid

**Review by ilya-lavrenov — APPROVED** [review]

**Review by vladimir-paramuzov — APPROVED** [review]
LGTM. Please have green CI to get this merged.

**Review by ilyachur — APPROVED** [review]

**Comment by olpipi** [comment]
rebased to latest master

**Pull request merged**